From: "Everest K.C." <everestkc@everestkc.com.np>
Subject: mm/execmem: remove logically deadcode in execmem.c
Date: Thu, 10 Oct 2024 12:10:58 -0600

NULL check of variable `area` within the `mas_for_each` loop is
unnecessary, as the varialbe `area` can never be NULL.  So, the `continue`
statement inside the if block is never reached.

Remove the if block that performs the NULL check.

This was reported by Coverity Scan:
https://scan7.scan.coverity.com/#/project-view/51525/11354?selectedIssue=1600362

Link: https://lkml.kernel.org/r/20241010181102.5522-1-everestkc@everestkc.com.np
Fixes: d44c3485820e ("execmem: add support for cache of large ROX pages")
Signed-off-by: Everest K.C. <everestkc@everestkc.com.np>
Cc: Kees Bakker <kees@ijzerbout.nl>
Cc: Mike Rapoport (Microsoft) <rppt@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/execmem.c |    3 ---
 1 file changed, 3 deletions(-)

--- a/mm/execmem.c~execmem-add-support-for-cache-of-large-rox-pages-fix
+++ a/mm/execmem.c
@@ -75,9 +75,6 @@ static void execmem_cache_clean(struct w
 	mas_for_each(&mas, area, ULONG_MAX) {
 		size_t size;
 
-		if (!area)
-			continue;
-
 		size = mas_range_len(&mas);
 
 		if (IS_ALIGNED(size, PMD_SIZE) &&
_
